src/infrastructure/apiRequest.ts

```diff
@@ -106,7 +106,7 @@ function prepareFetchOptions(requestConfig: RequestConfig, headers: Record<strin
 
   // Add body for non-GET requests
   if (requestConfig.data && requestConfig.method && requestConfig.method.toUpperCase() !== "GET") {
-    if (typeof requestConfig.data === "string") {
+    if (typeof requestConfig.data === "string" || requestConfig.data instanceof FormData) {
```

Comment:
Do not stringify `FormData` since `fetch` will handle it appropriately, including adding the necessary content headers